src/compiler/checker.rs

```diff
@@ -1026,12 +2465,30 @@ impl NodeBuilder {
         })
     }
 
-    fn with_context<TReturn, TCallback: FnMut(&NodeBuilderContext) -> TReturn>(
+    pub fn symbol_to_expression(
         &self,
+        type_checker: &TypeChecker,
+        symbol: Rc<Symbol>,
+        meaning: /*SymbolFlags*/ Option<SymbolFlags>,
         tracker: Option<&dyn SymbolTracker>,
-        mut cb: TCallback,
+    ) -> Option<Expression> {
+        self.with_context(tracker, |context| {
+            self._symbol_to_expression(type_checker, symbol, context, meaning)
+        })
+    }
+
+    fn with_context<TReturn, TCallback: FnOnce(&NodeBuilderContext) -> TReturn>(
+        &self,
+        tracker: Option<&dyn SymbolTracker>,
+        cb: TCallback,
     ) -> Option<TReturn> {
-        let context = NodeBuilderContext::new(tracker.unwrap());
+        let default_tracker: Option<DefaultNodeBuilderContextSymbolTracker> = if tracker.is_some() {
+            None
+        } else {
+            Some(DefaultNodeBuilderContextSymbolTracker::new())
+        };
```

Comment:
A `match` seems like the more idiomatic thing to do here, to me.

Reply:
Ok updated to use `match`